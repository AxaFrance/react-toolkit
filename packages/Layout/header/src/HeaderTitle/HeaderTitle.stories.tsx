import React from 'react';
import { action } from '@storybook/addon-actions';
import { Meta } from '@storybook/react';
import Action from '@axa-fr/react-toolkit-action';
import HeaderTitle from './HeaderTitle';
import readme from '../../README.md';

export default {
  title: 'Structure/Header/HeaderTitle',
  component: HeaderTitle,
  parameters: {
    readme: {
      sidebar: readme,
    },
    options: {},
  },
} as Meta;

const actions = [
  {
    id: '001',
    href: '#enregistrer',
    icon: 'floppy-disk',
    title: 'Enregistrer',
  },
  {
    id: '002',
    href: '#imprimer',
    icon: 'print',
    title: 'Imprimer',
  },
];

export const Default = (args) => <HeaderTitle {...args} />;
Default.args = {
  title: 'Toolkit Axa',
  subtitle: 'Info complÃ©mentaire',
  isSticky: true,
};
Default.argTypes = {
  toggleMenu: { action: 'onToggle' },
};

export const Complex = (args) => (
  <HeaderTitle {...args}>
    <div className="af-title-bar__actions">
      <a className="af-title-bar__link" href="#lien" title="lien titlebar">
        lien titlebar
      </a>
      {actions.map(({ icon, href, title, id }) => (
        <Action
          key={id}
          icon={icon}
          href={href}
          title={title}
          onClick={() => {
            action('Action click');
          }}
        />
      ))}
    </div>
  </HeaderTitle>
);
Complex.args = {
  ...Default.args,
};
Complex.argTypes = {
  ...Default.argTypes,
};
