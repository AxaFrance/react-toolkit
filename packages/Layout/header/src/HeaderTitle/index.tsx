export { default as HeaderTitle } from './HeaderTitle';
