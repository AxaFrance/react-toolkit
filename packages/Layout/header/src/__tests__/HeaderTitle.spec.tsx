import React from 'react';
import { render } from '@testing-library/react';
import { HeaderTitle } from '../HeaderTitle/index';

describe('<HeaderTitle>', () => {
  it('renders HeaderTitle correctly', () => {
    const { asFragment } = render(
      <HeaderTitle
        title="Toolkit Axa"
        subtitle="Info complémentaire"
        toggleMenu={() => null}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders HeaderTitle correctly without menu', () => {
    const { asFragment } = render(
      <HeaderTitle title="Toolkit Axa" subtitle="Info complémentaire" />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders HeaderTitle with classModifier', () => {
    const { asFragment } = render(
      <HeaderTitle
        title="Toolkit Axa"
        subtitle="Info complémentaire"
        classModifier="test"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
