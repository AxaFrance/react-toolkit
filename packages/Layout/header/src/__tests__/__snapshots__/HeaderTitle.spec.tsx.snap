// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderTitle> renders HeaderTitle correctly 1`] = `
<DocumentFragment>
  <div
    class="af-title-bar af-title-bar--sticky"
  >
    <div
      class="container af-title-bar__wrapper"
    >
      <div
        class="burger-container"
      >
        <a
          aria-controls="mainmenu"
          aria-haspopup="true"
          class="btn af-title-bar__mobile-menu af-btn--circle"
          href="/#"
          id="togglemenu"
          role="button"
          tabindex="0"
          title="Toggle menu"
        >
          <i
            class="glyphicon glyphicon-menu-hamburger"
          />
        </a>
      </div>
      <h1
        class="af-title-bar__title"
      >
        Toolkit Axa
        <span
          class="af-title-bar__subtitle"
        >
          Info complémentaire
        </span>
      </h1>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<HeaderTitle> renders HeaderTitle correctly without menu 1`] = `
<DocumentFragment>
  <div
    class="af-title-bar af-title-bar--sticky"
  >
    <div
      class="container af-title-bar__wrapper"
    >
      <h1
        class="af-title-bar__title"
      >
        Toolkit Axa
        <span
          class="af-title-bar__subtitle"
        >
          Info complémentaire
        </span>
      </h1>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<HeaderTitle> renders HeaderTitle with classModifier 1`] = `
<DocumentFragment>
  <div
    class="af-title-bar af-title-bar--test af-title-bar--sticky"
  >
    <div
      class="container af-title-bar__wrapper"
    >
      <h1
        class="af-title-bar__title"
      >
        Toolkit Axa
        <span
          class="af-title-bar__subtitle"
        >
          Info complémentaire
        </span>
      </h1>
    </div>
  </div>
</DocumentFragment>
`;
